src/util/lineItems.js

```diff
@@ -0,0 +1,168 @@
+  const numericPrice = convertMoneyToNumber(unitPrice);
+  const numericTotalPrice = new Decimal(numericPrice).times(unitCount).toNumber();
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice, unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+const calculateTotalPriceFromPercentage = (unitPrice, percentage) => {
+  const numericPrice = convertMoneyToNumber(unitPrice);
+  const numericTotalPrice = new Decimal(numericPrice)
+    .times(percentage)
+    .dividedBy(100)
+    .toNumber();
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice, unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+const calculateTotalPriceFromSeats = (unitPrice, unitCount, seats) => {
+  const numericPrice = convertMoneyToNumber(unitPrice);
+  const numericTotalPrice = new Decimal(numericPrice)
+    .times(unitCount)
+    .times(seats)
+    .toNumber();
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice, unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+export const validLineItem = params => {
+  const { code, unitPrice, quantity, percentage, seats, units, includeFor } = params;
+
+  let pricingParams;
+
+  if (quantity) {
+    pricingParams = { quantity, lineTotal: calculateTotalPriceFromQuantity(unitPrice, quantity) };
+  } else if (percentage) {
+    pricingParams = {
+      percentage,
+      lineTotal: calculateTotalPriceFromPercentage(unitPrice, percentage),
+    };
+  } else if (seats && units) {
+    pricingParams = {
+      seats,
+      units,
+      lineTotal: calculateTotalPriceFromSeats(unitPrice, units, seats),
+    };
+  } else {
+    console.error(
+      "Can't calculate the lineTotal of lineItem: ",
+      code,
+      ' Make sure you have provided quantity, percentage or both seats and units'
+    );
+  }
+
+  return {
+    code: code,
+    unitPrice,
+    includeFor: includeFor ? includeFor : ['customer', 'provider'],
+    ...pricingParams,
+  };
+};
+
+export const calculateTotalFromLineItems = lineItems => {
+  const numericTotalPrice = lineItems.reduce((sum, lineItem) => {
+    const numericPrice = convertMoneyToNumber(lineItem.lineTotal);
+    return new Decimal(numericPrice).add(sum);
+  }, 0);
+
+  const unitPrice = lineItems[0].unitPrice;
+
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice.toNumber(), unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+export const calculateTotalForProvider = lineItems => {
+  const providerLineItems = lineItems.filter(lineItem => lineItem.includeFor.includes('provider'));
+  return calculateTotalFromLineItems(providerLineItems);
+};
+
+export const calculateTotalForCustomer = lineItems => {
+  const providerLineItems = lineItems.filter(lineItem => lineItem.includeFor.includes('customer'));
+  return calculateTotalFromLineItems(providerLineItems);
+};
+
+export const constructLineItems = (startDate, endDate, unitPrice) => {
```

Comment:
Since customizations likely add new booking params, it would make sense to pass around a generic `bookingData` object that is used to construct the line items for the estimation and for the actual tx initiation in the backend. This way adding a new field to the booking form doesn't require changing all the places that pass around the booking data.